rfcs/065-library-data-link-explorer/README.md

```diff
@@ -0,0 +1,129 @@
+
+However, CI colleagues currently have no visibility over how records are linked other than by manually inspecting individual records. Since 2021, our team has maintained a script for visualising the way records are linked (we call these "matcher graphs" after the name of the pipeline service which identifies the links), but this script can only be run by software engineers due to the AWS permissions it requires. Since the source information belongs to Collections Information, we would like them to be able to view these graphs (nodes and edges) themselves.
+
+### How will it work?
+
+
+1. **User inputs a work ID**: A Collections Information staff member enters an work ID into the application.
+2. **API fetches the data**: The API queries Elasticsearch for records linked to the work ID.
+3. **Frontend displays the graph**: The frontend renders a visual graph of nodes (works) and edges (relationships), allowing users to explore the connections using a graph library.
+4. **Enhanced Collections Information workflow**: This allows Collections Information colleagues to independently view and explore relationships between records without requiring software engineer assistance, boosting productivity and collaboration.
+
+---
+
+## Backend
+
+- An API endpoint, powered by AWS Lambda to:
+    - query Elasticsearch for works, returning matched works of the given work id
+    - generate a DOT notation graph for this given work id using matcher logic within the endpoint
+    - transform this DOT notation into JSON ready for front end consumption
+- Accessed securely using AWS Secrets Manager for credentials for Elasticsearch
+
+### Helpers
+
+- Use the build of the content-api as an example API endpoint
+- JSON conversion is not strictly necessary, but useful for better graph libraries
+- [existing matcher logic](https://github.com/wellcomecollection/catalogue-pipeline/blob/main/pipeline/matcher_merger/matcher/scripts/getMatcherGraph.ts)
+
+### Technical Considerations
+
+- Frameworks and Dependencies
+    - TypeScript
+    - @elastic/elasticsearch client
+    - aws-lambda
+    - AWS SDK
+
+---
+
+## Frontend
+
+- A Next.js application that:
+    - provides a user interface for CI staff to enter a work id
+    - renders the matcher graph dynamically using a graph library (see candidates below)
+        - uses nodes and edges to show relationships
+    - handle error display when no relationships are present?
+       - potentially offer more advanced features than is currently possible (i.e, zoom, node highlighting)
+- Accessed securely - consider later down the line
+
+---
+
+## Infrastructure and Deployment
+
+- Use GitHub Actions for CI/CD to deploy the backend and frontend
+- keeping repositories separate for clarity and modularity between the API/backend logic and frontend application and deployment
```

Comment:
Are you planning on having a repository for the Lambda & another for the frontend, or just one? It's probably unnecessary to have two repositories.

Reply:
I was hoping one would be the best case, but naively thought it might *have* to be two. So will stick with one!